File: Backend/RaiseYourVoice.Domain/Entities/Comment.cs
using RaiseYourVoice.Domain.Common;

namespace RaiseYourVoice.Domain.Entities
{
    public class Comment : BaseEntity
    {
        public required string PostId { get; set; }
        public required string AuthorId { get; set; }
        public required string Content { get; set; }
        public int LikeCount { get; set; }
        public string? ParentCommentId { get; set; }

        public bool IsReported { get; set; } = false;
        public string? ReportedBy { get; set; }
        public DateTime? ReportResolvedAt { get; set; }
    }
}

File: Mobile/Android/app/src/main/java/al/raiseyourvoice/android/domain/model/Comment.kt
package al.raiseyourvoice.android.domain.model

import java.util.Date

/**
 * Domain model representing a comment in the application.
 */
data class Comment(
    val id: String,
    val postId: String,
    val authorId: String,
    val authorName: String,
    val authorProfilePicUrl: String? = null,
    val content: String,
    val createdAt: Date,
    val updatedAt: Date? = null,
    val likeCount: Int = 0,
    val isLikedByCurrentUser: Boolean = false,
    val parentCommentId: String? = null,
    val childCommentCount: Int = 0
)